### DISCUSSION TITLE
Use of Oauth2 Google external (TortoiseBaseOAuthAccountModel) and TortoiseBaseUserModel

### DISCUSSION THREAD

**Discussion by diegocgaona** [fastapi-users/fastapi-users#618] [Q&A] [answered]
Hi,

I'm thinking of use the authentication for user with e-mail and password, the normal flow, and at same time the option to authentication with Google sign in.

I'm using Tortoise, and I have class `UserModel(TortoiseBaseUserModel)`, but, the Oauth2 with Google use another model `OAuthAccountModel(TortoiseBaseOAuthAccountModel)`.

How can I make the two work together? I need two tables? Or I can create optional fields in `TortoiseBaseUserModel`?

Thanks!!

**Comment by frankie567** [comment, marked as answer]
Yes, you need both. You'll have two tables:

* One to store the users
* One to store OAuth credentials. It has a foreign key linking to a user.

You have a complete example here: https://frankie567.github.io/fastapi-users/configuration/oauth/#tortoise-orm_1

**Comment by diegocgaona** [comment]
Thank you very much!!
I didn't saw that the example have both working together.
